components/ui/PortalPopup.tsx

```diff
@@ -0,0 +1,228 @@
+  throttledPositionTooltip: () => void;
+
+  constructor(props: Props) {
+    super(props);
+    this.state = {};
+    this.throttledPositionTooltip = throttle(this.positionPopup, 10);
+  }
+
+  componentDidMount() {
+    this.mounted = true;
+    this.positionPopup();
+    this.addEventListeners();
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (this.props.placement !== prevProps.placement || this.props.overlay !== prevProps.overlay) {
+      this.positionPopup();
+    }
+  }
+
+  componentWillUnmount() {
+    this.mounted = false;
+    this.removeEventListeners();
+  }
+
+  addEventListeners = () => {
+    window.addEventListener('resize', this.throttledPositionTooltip);
+    window.addEventListener('scroll', this.throttledPositionTooltip);
+  };
+
+  removeEventListeners = () => {
+    window.removeEventListener('resize', this.throttledPositionTooltip);
+    window.removeEventListener('scroll', this.throttledPositionTooltip);
+  };
+
+  getPlacement = (): PopupPlacement => {
+    return this.props.placement || PopupPlacement.Bottom;
+  };
+
+  adjustArrowPosition = (
+    placement: PopupPlacement,
+    { leftFix, topFix }: { leftFix: number; topFix: number }
+  ) => {
+    const { arrowOffset = 0 } = this.props;
+    switch (placement) {
+      case PopupPlacement.Bottom:
+      case PopupPlacement.BottomLeft:
+      case PopupPlacement.BottomRight:
+      case PopupPlacement.TopLeft:
+        return { marginLeft: -leftFix + arrowOffset };
+      default:
+        return { marginTop: -topFix + arrowOffset };
+    }
+  };
+
+  positionPopup = () => {
+    // `findDOMNode(this)` will search for the DOM node for the current component
+    // first it will find a React.Fragment (see `render`),
+    // so it will get the DOM node of the first child, i.e. DOM node of `this.props.children`
+    // docs: https://reactjs.org/docs/refs-and-the-dom.html#exposing-dom-refs-to-parent-components
+
+    // eslint-disable-next-line react/no-find-dom-node
+    const toggleNode = findDOMNode(this);
+
+    if (toggleNode && toggleNode instanceof Element && this.popupNode.current) {
+      const toggleRect = toggleNode.getBoundingClientRect();
+      const { width, height } = this.popupNode.current.getBoundingClientRect();
+      let left = 0;
+      let top = 0;
+
+      switch (this.getPlacement()) {
+        case PopupPlacement.Bottom:
+          left = toggleRect.left + toggleRect.width / 2 - width / 2;
+          top = toggleRect.top + toggleRect.height;
+          break;
+        case PopupPlacement.BottomLeft:
+          left = toggleRect.left;
+          top = toggleRect.top + toggleRect.height;
+          break;
+        case PopupPlacement.BottomRight:
+          left = toggleRect.left + toggleRect.width - width;
+          top = toggleRect.top + toggleRect.height;
+          break;
+        case PopupPlacement.LeftTop:
+          left = toggleRect.left - width;
+          top = toggleRect.top;
+          break;
```

Comment:
Not sure all positionings are correct. See this for `LeftTop`:

<img width="480" alt="Screenshot 2020-07-09 at 10 46 35" src="https://user-images.githubusercontent.com/45544358/87018464-91671900-c1d1-11ea-82ac-d6459c2d3758.png">

Could be due to the surrounding elements, which might confuse `ScreenPositionFixer`, but it's worth to double check.

Reply:
I just checked again and I think it's correct. Here too, the top of the popup is on the left and is aligned with the top of your text. You'd get the same result if you used a normal popup.